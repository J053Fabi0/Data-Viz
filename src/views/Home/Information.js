import React from "react";
import { Table } from "react-bootstrap";
const { states } = require("../../utils/constants");

export default function Informaci√≥n({ data, selectedStateIndex }) {
  const { minIDH, maxIDH, averageIDH } = data[selectedStateIndex].data;

  return (
    <div className="data mt-2">
      <Table striped bordered>
        <thead>
          <tr>
            <th colSpan="2">IDH de {states.names[selectedStateIndex]}</th>
          </tr>
        </thead>

        <tbody>
          <tr>
            <td>Promedio</td>
            <td>{averageIDH}</td>
          </tr>
          <tr>
            <td>Menor</td>
            <td>{minIDH}</td>
          </tr>
          <tr>
            <td>Mayor</td>
            <td>{maxIDH}</td>
          </tr>
        </tbody>
      </Table>
    </div>
  );
}
