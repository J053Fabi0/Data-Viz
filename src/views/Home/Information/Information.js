import React from "react";
import { Table } from "react-bootstrap";
const { states } = require("../../../utils/constants");

export default function Informaci√≥n({ data, selectedStateIndex }) {
  const { minIDH, maxIDH, averageIDH } = data[selectedStateIndex].data;

  return (
    <div className="data mt-2" data-testid="information">
      <Table striped bordered>
        <thead>
          <tr>
            <th data-testid="title" colSpan="2">
              IDH de {states.names[selectedStateIndex]}
            </th>
          </tr>
        </thead>

        <tbody>
          <tr>
            <td>Promedio</td>
            <td data-testid="average">{averageIDH}</td>
          </tr>
          <tr>
            <td>Menor</td>
            <td data-testid="min">{minIDH}</td>
          </tr>
          <tr>
            <td>Mayor</td>
            <td data-testid="max">{maxIDH}</td>
          </tr>
        </tbody>
      </Table>
    </div>
  );
}
