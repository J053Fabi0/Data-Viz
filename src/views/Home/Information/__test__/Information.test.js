import React from "react";
import Information from "../Information";

import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";

const data = [
  {
    data: {
      minIDH: 0.1,
      averageIDH: 0.5,
      maxIDH: 1,
    },
  },
];
const selectedStateIndex = 0;

test("Matches snapshot", () => {
  const tree = renderer.create(<Information data={data} selectedStateIndex={selectedStateIndex} />).toJSON();
  expect(tree).toMatchSnapshot();
});
