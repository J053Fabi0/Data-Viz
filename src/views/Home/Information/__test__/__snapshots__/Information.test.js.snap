// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="data mt-2"
  data-testid="information"
>
  <table
    className="table table-striped table-bordered"
  >
    <thead>
      <tr>
        <th
          colSpan="2"
        >
          IDH de 
          Aguascalientes
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          Promedio
        </td>
        <td>
          0.5
        </td>
      </tr>
      <tr>
        <td>
          Menor
        </td>
        <td>
          0.1
        </td>
      </tr>
      <tr>
        <td>
          Mayor
        </td>
        <td>
          1
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
